src/createInverseContext.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+
+export const createInverseContext = <T extends unknown = null>(
+  initialValue: T,
+) => {
+  const Context = React.createContext(initialValue) as React.Context<T> & {
+    __updateProviderValue?: (value: T) => void;
+  };
+
+  class InverseConsumer extends React.Component {
+    state = { value: initialValue };
+    constructor(props: any) {
+      super(props);
+      Context.__updateProviderValue = (value: T) => this.setState({ value });
```

Comment:
why do we need a state here? will this not cause an extra rerender?

Reply:
extra in what sense? What line were you referring to exactly?

Reply:
I meant line 11 since the value is maintained in the state so wanted to know if that's needed.

Reply:
It's necessary so as to make the component rerender, which in turn updates the provider value and triggers consumer components render. There shouldn't be any extra renders from what I can see.